### DISCUSSION TITLE
Adding userInfo dictionary to IssueReporting?

### DISCUSSION THREAD

**Discussion by larryonoff** [pointfreeco/swift-issue-reporting#136] [Ideas]
I was planning to implement IssueReporting that uses `FirebaseCrashlytics.recordError()` with `userInfo: [String: Any]`, but IssueReporting currently lacks a `userInfo` parameter. What do you think about adding `userInfo`?

**Comment by lukeredpath** [comment]
Could you not just bundle this up in your error type? You could even use `NSError` or if you have custom error types already, you can conform them to `CustomNSError`.

Then in your custom issue reporter, you would just need to see if the error casts to that protocol and extract the user info.

**Reply by larryonoff** [reply]
Sounds like a great idea!
